File: solutions/fsharp/cars-assemble/2/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int): float =
    if speed <= 0 then
        0.0
    elif 1 <= speed && speed < 5 then
        1.0
    elif 5 <= speed && speed < 9 then
        0.9
    elif 9 <= speed && speed < 10 then
        0.8
    else
        0.77

let carsPerOneHour = 221

let productionRatePerHour (speed: int): float =
    successRate(speed) * float(speed * carsPerOneHour)

let workingItemsPerMinute (speed: int): int =
    productionRatePerHour(speed) / 60.0
        |> int

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;

static class AssemblyLine
{
    private const int CARS_PER_HOUR = 221;

    public static double SuccessRate(int speed)
    {
        switch (speed)
        {
            case <= 4 when speed > 0:
                return 1;
            case <= 8 when speed > 4:
                return 0.9;
            case 9:
                return 0.8;
            case 10:
                return 0.77;
            default:
                return 0;
        }
    }

    public static double ProductionRatePerHour(int speed) =>
        SuccessRate(speed) * CARS_PER_HOUR * speed;

    public static int WorkingItemsPerMinute(int speed) =>
        (int)(ProductionRatePerHour(speed) / 60);
}
